src/pages/workspaces/workspace/Workflows.js

```diff
@@ -415,15 +416,29 @@ export const Workflows = _.flow(
       h(ViewToggleButtons, { listView, setListView }),
       workflowToExport &&
         h(ExportWorkflowModal, {
-          thisWorkspace: workspace,
-          methodConfig: getConfig(workflowToExport),
+          defaultWorkflowName: getConfig(workflowToExport).name,
+          destinationWorkspace: ({ workspace: { workspaceId }, accessLevel }) => {
+            return workspace.workspaceId !== workspaceId && WorkspaceUtils.canWrite(accessLevel);
+          },
+          title: 'Copy to Workspace',
+          exportButtonText: 'Copy',
+          exportProvider: makeExportWorkflowFromWorkspaceProvider(workspace, getConfig(workflowToExport)),
```

Comment:
factory methods like this can and should be called outside of component render methods (functional components are just a render method).  The best option would be to have this actually come from a component prop so that the consumer can take care of deciding on the implementation (make call)... and yeah, that means it would be beneficial to do at least a partial convertion of Workflows.js to .ts so you can define component props.

Reply:
from our chat:

Looks like you have a legit reason to use a factory method since you will want divergence on the "source" arguments, so having them baked into the provider makes sense. This is a good design for your needs.  Nice work.